### ISSUE TITLE
Inferencing on a custom .gif after training the model

### ISSUE THREAD

**Issue by chpk** [lucidrains/video-diffusion-pytorch#29]
Hi everyone,

I have trained the model on a custom data, the milestones/checkpoints are saved in "./results" folder. The training code was straight forward, but during inference, i am not able to figure out how to infer on a custom input .gif using the saved checkpoint/milestones present in the "./results" folder.

Thanks in advance,

**Comment by kkedich** [comment]
Are you trying to use a GIF input file to make an inference? I think I did not understand well you request. If this is the case, I am not sure if this is available in the repository. But if you meant to sample a video after training, then you can use the `diffusion.sample` function presented in the README.md file

**Comment by chpk** [comment]
you can use the following code for sampling from trained model parameters:
import torch
from video_diffusion_pytorch import Unet3D, GaussianDiffusion, Trainer, video_tensor_to_gif

batch_size = 1

model = Unet3D(
    dim = 64,
    cond_dim  = 768,
    dim_mults = (1, 2, 4, 8),
)

diffusion = GaussianDiffusion(
    model,
    image_size = 64,
    num_frames = 10,
    channels = 3,
    timesteps = 1000,   # number of steps
    loss_type = 'l1'    # L1 or L2
)

trainer = Trainer(
    diffusion,
    './data')

trainer.load(-1)


input_condition = torch.randn(batch_size, 768) 

output_Gif_Tensor = diffusion.sample(cond = input_condition, batch_size = batch_size)

video_tensor_to_gif(output_Gif_Tensor[0], './output.gif')
